libraries/src/AWS.Lambda.Powertools.Common/Core/SystemWrapper.cs

```diff
@@ -78,4 +83,51 @@ public double GetRandom()
     {
         return new Random().NextDouble();
     }
+
+    /// <inheritdoc />
+    public void SetEnvironmentVariable(string variable, string value)
+    {
+        _powertoolsEnvironment.SetEnvironmentVariable(variable, value);
+    }
+
+    /// <inheritdoc />
+    public void SetExecutionEnvironment<T>(T type)
```

Comment:
I don't know the whole system so it might already being doing this but make sure this is only called once per power tools assembly. These assembly look up calls are can be costly and we don't want to it for every request.

Reply:
Thanks for the feedback @normj . The implementation prevents multiple instances of SystemWrapper to be created. The instances that create systemwrapper are singletons. The constructors are called once for the lambda lifetime.